src/garage/torch/algos/maml/maml_base.py

```diff
@@ -0,0 +1,487 @@
+
+        Args:
+            task_samples (list[MAMLTrajectoryBatch]): Samples data for
+                one task.
+
+        Returns:
+            torch.Tensor: Computed entropy value.
+
+        """
+        # pylint: disable=protected-access
+        entropies = [
+            self._inner_algo._compute_policy_entropy(samples.observations)
+            for samples in task_samples
+        ]
+        return torch.stack(entropies).mean()
+
+    def _set_task(self, runner, task):
+        # pylint: disable=protected-access, no-self-use
+        for env in runner._sampler._vec_env.envs:
+            env.set_task(task)
+
+    def _set_baseline(self, task_id):
+        self._cur_baseline = self._baselines[task_id]
+        self._inner_algo.baseline = self._cur_baseline
+
+    @property
+    def policy(self):
+        """Current policy of the inner algorithm.
+
+        Returns:
+            garage.torch.policies.Policy: Current policy of the inner
+                algorithm.
+
+        """
+        return self._policy
+
+    @property
+    def _old_policy(self):
+        """Old policy of the inner algorithm.
+
+        Returns:
+            garage.torch.policies.Policy: Old policy of the inner algorithm.
+
+        """
+        # pylint: disable=protected-access
+        return self._inner_algo._old_policy
+
+    def _process_samples(self, itr, paths):
+        """Process sample data based on the collected paths.
+
+        Args:
+            itr (int): Iteration number.
+            paths (list[dict]): A list of collected paths
+
+        Returns:
+            MAMLTrajectoryBatch: Processed samples data.
+
+        """
+        for path in paths:
+            path['returns'] = tensor_utils.discount_cumsum(
+                path['rewards'], self._inner_algo.discount)
+
+        self._cur_baseline.fit(paths)
+        obs, actions, rewards, valids, baselines \
+            = self._inner_algo.process_samples(itr, paths)
+        return MAMLTrajectoryBatch(obs, actions, rewards, valids, baselines)
+
+    def evaluate_performance(self, itr, all_samples, loss_before, loss_after,
+                             kl_before, kl, policy_entropy):
+        """Evaluate performance of this batch.
+
+        Args:
+            itr (int): Iteration number.
+            all_samples (list[list[MAMLTrajectoryBatch]]): Two
+                dimensional list of MAMLTrajectoryBatch of size
+                [meta_batch_size * (num_grad_updates + 1)]
+            loss_before (float): Loss before optimization step.
+            loss_after (float): Loss after optimization step.
+            kl_before (float): KL divergence before optimization step.
+            kl (float): KL divergence after optimization step.
+            policy_entropy (float): Policy entropy.
+
+        Returns:
+            float: The average return in last epoch cycle.
+
+        """
+        tabular.record('Iteration', itr)
+
+        for i in range(self._num_grad_updates + 1):
```

Comment:
Is there a way to just use `RLAlgorithm.evaluate_performance`?

Reply:
I don't think so because it's the post-adaptation performance that matters, and the performance number is averaged over tasks.

Reply:
or `MetaRLAlgorithm.evaluate_train_performance` and `MetaRLAlgorithm.evaluate_test_performance`?